File: minsizesubarraysum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left=0,sum=0;
        int minlen=numeric_limits<int>::max();
        for(int r=0;r<nums.size();r++)
        {
            sum=sum+nums[r];
            while(sum>=target)
            {
                if(minlen>r-left+1)
                {
                    minlen=r-left+1;
                }
                sum=sum-nums[left];
                left++;
            }
        }
        return minlen!=numeric_limits<int>::max()?minlen:0;
    }
};

File: minsizesubarraysum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        minlen=float("inf")
        left=0
        sum=0
        for i in range(len(nums)):
            sum+=nums[i]
            while sum>=target:
                if minlen>i-left+1:
                    minlen=i-left+1
                sum-=nums[left]
                left=left+1
        return minlen if minlen!=float("inf") else 0
        
